cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/DefaultWorkflowConfigurer.java

```diff
@@ -112,8 +139,7 @@ private WorkflowNode createNodeWithId(WorkflowNode node) {
     WorkflowNode nodeWithId = null;
     switch (node.getType()) {
       case ACTION:
-        nodeWithId = createActionNodeWithId(node);
-        break;
+        return node;
```

Comment:
So now some nodes will have explicit unique names, given by the developer, and others have generated numbers as the unique id?